src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -59,63 +53,77 @@ final class SettingsTable extends AbstractVirtualTable
                            .addPartitionKeyColumn(NAME, UTF8Type.instance)
                            .addRegularColumn(VALUE, UTF8Type.instance)
                            .build());
-        this.config = config;
+        Set<String> unprocessed = new HashSet<>(BACKWARDS_COMPATABLE_NAMES.values());
+        DatabaseDescriptor.visit((key, type, ro) -> {
+            if (BACKWARDS_COMPATABLE_NAMES.containsKey(key))
+            {
+                throw new AssertionError(String.format("Name '%s' is present in Config, this adds a conflict as this " +
+                                                       "name had a different meaning in the '%s",
+                                                       key, SettingsTable.class.getSimpleName()));
+            }
+            unprocessed.remove(key);
+        });
+        if (!unprocessed.isEmpty())
+            throw new AssertionError("The following keys must be present in the Config: " + unprocessed);
```

Comment:
If you add `DatabaseDescriptor.hasProperty(String name)` you can drastically simplify this code...

```
for (String name : BACKWARDS_COMPATABLE_NAMES)
  if (DatabaseDescriptor.hasProperty(name))
    throw new AssertionError(String.format("Name '%s' is present in Config, this adds a conflict as this " +
                                                        "name had a different meaning in the '%s",
                                                        key, SettingsTable.class.getSimpleName()));
```

This is also cheaper

Reply:
Fixed.